skimage/morphology/_flood_fill.py

```diff
@@ -271,16 +272,19 @@ def flood(image, seed_point, *, footprint=None, connectivity=None, tolerance=Non
 
     try:
         if tolerance is not None:
-            # Check if tolerance could create overflow problems
-            try:
-                max_value = np.finfo(working_image.dtype).max
-                min_value = np.finfo(working_image.dtype).min
-            except ValueError:
-                max_value = np.iinfo(working_image.dtype).max
-                min_value = np.iinfo(working_image.dtype).min
-
-            high_tol = min(max_value, seed_value + tolerance)
-            low_tol = max(min_value, seed_value - tolerance)
+            tolerance = abs(tolerance)
+            # Account for over- & underflow problems with seed_value ± tolerance
+            # in a way that works with NumPy 1 & 2
+            min_value, max_value = numeric_dtype_min_max(seed_value.dtype)
+            with np.errstate(over="raise", under="raise"):
+                try:
+                    low_tol = max(min_value, seed_value - tolerance)
+                except (OverflowError, FloatingPointError):
+                    low_tol = min_value
+                try:
+                    high_tol = min(max_value, seed_value + tolerance)
+                except (OverflowError, FloatingPointError):
+                    high_tol = max_value
```

Comment:
Some explanation: in NumPy 2.0 we can no longer rely on 
```python
high_tol = min(max_value, seed_value + tolerance)
low_tol = max(min_value, seed_value - tolerance)
```
to correctly bound to the min and max of the dtype. `seed_value` may be a `np.uint8` in which case adding e.g. 379 to it will result in an OverflowError. 

`with np.errstate(over="raise", under="raise")` is necessary because in NumPy 2 cases like
```python
low_tol = max(0, np.uint8(2) - 3)
```

Though, I may be wrong. I find it very tricky to think and test for all edge cases...
now underflow to `np.uint8(255)` because the 3 will be cast to np.uint8 before addtion. In NumPy 1, the right side would result in np.int64(-1).

Reply:
Yes, but you will get a RuntimeWarning, which should make our test suite error out.

Easiest in this case may be `max(0, int(x) - 3)`.

Reply:
Casting to the appropriate Python scalar seems indeed like the easiest fix. :+1: